### PR TITLE
[do not merge] Dynamic provisioning admin guide

### PR DESCRIPTION
**Pull request by markturansky**
First draft doc to support https://github.com/openshift/origin/pull/5217

Upstream README here:  https://github.com/kubernetes/kubernetes/pull/14537

### CODE REVIEW

File: admin_guide/provisioning-pvs.adoc [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+The plugin names are:
+
+1. OpenStack Cinder - "kubernetes.io/cinder"
+2. AWS EBS - "kubernetes.io/aws-ebs"
+3. GCEPD - "kubernetes.io/gce-pd"
+
+Future provisioners will include the many types of storage a single provider offers.  AWS, for example, has several types of EBS volumes to offer,
+each with its own performance characteristics.  They also have an NFS-like storage option.  More provisioners will be implemented for our storage providers.
+
+[[Requesting storage]]
```
Comment (adellape) [inline admin_guide/provisioning-pvs.adoc]: Needs a hyphen: `[[requesting-storage]]`

File: admin_guide/provisioning-pvs.adoc [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      "ReadWriteOnce"
+    ],
+    "resources": {
+      "requests": {
+        "storage": "3Gi"
+      }
+    }
+  }
+}
+----
```
Comment (adellape) [inline admin_guide/provisioning-pvs.adoc]: Trying to stick with YAML for examples:

```
====
[source,yaml]
----
  kind: "PersistentVolumeClaim"
  apiVersion: "v1"
  metadata: 
    name: "claim1"
    annotations: 
      volume.experimental.kubernetes.io/quality-of-service: "foo"
  spec: 
    accessModes: 
      - "ReadWriteOnce"
    resources: 
      requests: 
        storage: "3Gi"
----
====
```

File: admin_guide/provisioning-pvs.adoc [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+====
+
+[[Reclaiming Resources]]
```
Comment (adellape) [inline admin_guide/provisioning-pvs.adoc]: Needs a hyphen: `[[reclaiming-resources]]`

**Comment by adellape** [comment]
@markturansky Thanks for this. A few comments inline. Marking `[do not merge]` for now, keeping an eye on the code PRs. Not sure what OSE version to target yet (3.1 or 3.1.z).

**Comment by markturansky** [comment]
thanks @adellape, this is now slated for 3.1.z.

**Comment by adellape** [comment]
@markturansky Looks like the referenced https://github.com/kubernetes/kubernetes/pull/14537 was merged but https://github.com/openshift/origin/pull/5217 was closed. Does this docs PR need any updating?

**Comment by markturansky** [comment]
I think this is a "yes but ..."

This feature was added to 3.1.1 so that OpenShift Online can make use of it.  I can help internal RH folks use provisioning, but it's labeled an Alpha feature and my thinking is to wait until the full feature drops in 3.2 to provide complete docs.  I'm not sure we want to advertise something that will likely change by 3.2.

**Comment by adellape** [comment]
Closing in favor of https://github.com/openshift/openshift-docs/pull/1538.

**Pull request closed without merging**